deps/picotls/include/picotls.h

```diff
@@ -1444,6 +1447,120 @@ extern PTLS_THREADLOCAL unsigned ptls_default_skip_tracing;
+    do {                                                                                                                           \
+        if (!ptlslog_is_active())                                                                                                  \
+            break;                                                                                                                 \
+        char smallbuf[128];                                                                                                        \
+        ptls_buffer_t ptlslogbuf;                                                                                                  \
+        ptls_buffer_init(&ptlslogbuf, smallbuf, sizeof(smallbuf));                                                                 \
+        int ptlslog_skip = 0;                                                                                                      \
+        PTLSLOG__DO_PUSH_SAFESTR("{\"module\":\"" PTLS_TO_STR(module) "\",\"type\":\"" PTLS_TO_STR(type) "\"");                    \
+        do {                                                                                                                       \
+            block                                                                                                                  \
+        } while (0);                                                                                                               \
+        PTLSLOG__DO_PUSH_SAFESTR("}\n");                                                                                           \
+        if (!ptlslog_skip) {                                                                                                       \
+            ptlslog__do_write(&ptlslogbuf);                                                                                        \
+        }                                                                                                                          \
+        ptls_buffer_dispose(&ptlslogbuf);                                                                                          \
+    } while (0)
+
+#define PTLSLOG_CONN(type, tls, block)                                                                                             \
+    PTLSLOG(picotls, type, {                                                                                                       \
+        ptls_t *_tls = (tls);                                                                                                      \
+        ptlslog_skip = ptls_skip_tracing(_tls);                                                                                    \
+        if (ptlslog_skip)                                                                                                          \
+            break;                                                                                                                 \
```

Comment:
I do not think we'd want to build the prefix (i.e., `{"module": ...`) then call out that the connection is not traced.

Hence maybe change the definition to:
```
    if (!ptls_skip_tracing(_tls)) {
        PTLS_LOG_CONN(module, type, {
            PTLSLOG_ELEMENT_PTR(tls, _tls);
            do {
                block
            } while (0);
        });
    }
```
or something alike?